manifests/config.pp

```diff
@@ -157,66 +157,58 @@
     # the vast majority of files here are init-files
     # so any change there should trigger a full service restart
     if $prometheus::server::restart_on_change {
-      File {
-        notify => Class['prometheus::run_service'],
-      }
-      $systemd_notify = [Exec['prometheus-systemd-reload'], Class['prometheus::run_service']]
+      $notify = Class['prometheus::run_service']
     } else {
-      $systemd_notify = Exec['prometheus-systemd-reload']
+      $notify = undef
     }
 
     case $prometheus::server::init_style {
       'upstart' : {
         file { '/etc/init/prometheus.conf':
+          ensure  => file,
           mode    => '0444',
           owner   => 'root',
           group   => 'root',
           content => template('prometheus/prometheus.upstart.erb'),
+          notify  => $notify,
         }
         file { '/etc/init.d/prometheus':
           ensure => link,
           target => '/lib/init/upstart-job',
           owner  => 'root',
           group  => 'root',
           mode   => '0755',
+          notify => $notify,
         }
       }
       'systemd' : {
-        include 'systemd'
         systemd::unit_file {'prometheus.service':
           content => template('prometheus/prometheus.systemd.erb'),
+          notify  => $notify,
         }
       }
-      'sysv', 'redhat' : {
-        file { '/etc/init.d/prometheus':
-          mode    => '0555',
-          owner   => 'root',
-          group   => 'root',
-          content => template('prometheus/prometheus.sysv.erb'),
-        }
-      }
-      'debian' : {
-        file { '/etc/init.d/prometheus':
-          mode    => '0555',
-          owner   => 'root',
-          group   => 'root',
-          content => template('prometheus/prometheus.debian.erb'),
+      'sysv', 'redhat', 'debian', 'sles' : {
```

Comment:
These all declare the same file, so I've tried to remove a bit of duplication here.